query-dsl/src/main/java/org/infinispan/query/dsl/impl/BaseQuery.java

```diff
@@ -35,6 +36,8 @@
 
    protected int maxResults;
 
+   protected Optional<Integer> hitCountAccuracy = Optional.empty();
```

Comment:
I personally prefer not using Optional as a field variable, but instead only wrap the return value for a method invocation. Thus I would suggest a NULL Integer or some sort of -1 or other placeholder to signify it isn't set.

Reply:
I'm applying the suggestion...